app/models/parser/twitter_item.rb

```diff
@@ -19,47 +19,63 @@ def patterns
     # Main function for class
     def parse_data_for_parser(_doc, _original_url, _jsonld_array)
       @url.gsub!(/(%23|#)!\//, '')
+      @url.gsub!(/\s/, '')
       @url = replace_subdomain_pattern(url)
+      
       parts = url.match(TWITTER_ITEM_URL)
       user, id = parts['user'], parts['id']
       
-      @parsed_data['raw']['api'] = {}
-      handle_twitter_exceptions do
-        @parsed_data['raw']['api'] = twitter_client.status(id, tweet_mode: 'extended').as_json
+      @parsed_data['raw']['api'] = tweet_lookup(id)
+
+      @parsed_data[:error] = parsed_data['raw']['api']['errors']
+
+      if @parsed_data[:error] 
+        title = url
+        description = ''
+        author_picture = ''
+        published_at = ''
+        html = ''
+        author_name = user
+        author_url = get_author_url(user)
+      elsif @parsed_data[:error].nil?
+        raw_data = parsed_data['raw']['api']['data'][0]
```

Comment:
**suggestion**: for accessing deeply nested data where there might not always be a value, I'd suggest using dig, ie. `parsed_data.dig('raw','api','data',0)`. this will return nil if data is missing, whereas the current line and some others below would raise an error for invalid access.